examples/js/loaders/XfileLoader.js

```diff
@@ -0,0 +1,1518 @@
+
+    this.boneName = "";
+    this.BoneIndex = 0;
+    this.Indeces = [];
+    this.Weights = [];
+    this.initMatrix = null;
+    this.OffsetMatrix = null;
+
+};
+
+var XAnimationInfo$1 = XAnimationInfo = function XAnimationInfo() {
+
+    this.animeName = "";
+    this.boneName = "";
+    this.targetBone = null;
+    this.frameStartLv = 0;
+    this.keyFrames = [];
+    this.InverseMx = null;
+
+};
+
+var XFrameInfo$1 = XFrameInfo = function XFrameInfo() {
+
+    this.Mesh = null;
+    this.Geometry = null;
+    this.FrameName = "";
+    this.ParentName = "";
+    this.frameStartLv = 0;
+    this.FrameTransformMatrix = null;
+    this.children = [];
+    this.BoneInfs = [];
+    this.VertexSetedBoneCount = [];
+    this.Materials = [];
+
+};
+
+var XKeyFrameInfo = function XKeyFrameInfo() {
+
+    classCallCheck(this, XKeyFrameInfo);
+
+    this.index = 0;
+    this.Frame = 0;
+    this.time = 0.0;
+    this.matrix = null;
+
+};
+
+THREE.XFileLoader = function () {
+
+    function XFileLoader(manager, Texloader, _zflg) {
+
+        classCallCheck(this, XFileLoader);
+
+        this.manager = manager !== undefined ? manager : THREE.DefaultLoadingManager;
+        this.Texloader = Texloader !== undefined ? Texloader : THREE.TextureLoader;
+        this.zflg = _zflg === undefined ? false : _zflg;
+        this.skinFlg = false;
+        this.url = "";
+        this.baseDir = "";
+        this.nowReadMode = XfileLoadMode$1.none;
+        this.nowAnimationKeyType = 4;
+        this.tgtLength = 0;
+        this.nowReaded = 0;
+        this.elementLv = 0;
+        this.geoStartLv = Number.MAX_VALUE;
+        this.frameStartLv = Number.MAX_VALUE;
+        this.matReadLine = 0;
+        this.putMatLength = 0;
+        this.nowMat = null;
+        this.BoneInf = new XboneInf();
+        this.tmpUvArray = [];
+        this.normalVectors = [];
+        this.facesNormal = [];
+        this.nowFrameName = "";
+        this.nowAnimationSetName = "";
+        this.frameHierarchie = [];
+        this.endLineCount = 0;
+        this.geometry = null;
+        this.loadingXdata = null;
+        this.lines = null;
+        this.keyInfo = null;
+        this.animeKeyNames = null;
+        this.data = null;
+        this.onLoad = null;
+
+    }
+
+    createClass(XFileLoader, [{
+        key: 'load',
+        value: function load(_arg, onLoad, onProgress, onError) {
+
+            var _this = this;
+
+            var loader = new THREE.FileLoader(this.manager);
+            loader.setResponseType('arraybuffer');
+            for (var i = 0; i < _arg.length; i++) {
+
+                switch (i) {
+
+                    case 0:
+                        this.url = _arg[i]; break;
+                    case 1:
+                        this.zflg = _arg[i]; break;
+                    case 2:
+                        this.skinFlg = _arg[i]; break;
+
+                }
+
+            }
+            loader.load(this.url, function (text) {
```

Comment:
It will cause confusion and will change. As you pointed out, we decide whether it is text or Arraybuffer in parser.